### PR TITLE
Django

### PR DESCRIPTION
**Pull request by fhrr-sht**

### CODE REVIEW

**Comment by mezgoodle** [comment]
[Template](https://getbootstrap.com/docs/5.0/examples/sign-in/) needs to set

**Review by mezgoodle — APPROVED** [review]

File: server/mesite/models.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class Song(models.Model):
+    """
+    Class that represents Song object
+    fields: name, artist, id, acousticness, danceability, energy, instrumentalness,
+    liveness, valence, loudness, speechiness, tempo, key, time_signature, mood
+    """
+    name = models.CharField('Name', blank=False, max_length=40)
+    artist = models.CharField('Artists', blank=False, max_length=20)
```
Comment (mezgoodle) [inline server/mesite/models.py:32]:
```suggestion
    artists = models.CharField('Artists', blank=False, max_length=20)
```

File: server/mesite/models.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    acousticness = models.FloatField()
+    danceability = models.FloatField()
+    energy = models.FloatField()
+    instrumentalness = models.FloatField()
+    liveness = models.FloatField()
+    valence = models.FloatField()
+    loudness = models.FloatField()
+    speechiness = models.FloatField()
+    tempo = models.FloatField()
+    key = models.IntegerField()
```
Comment (mezgoodle) [inline server/mesite/models.py:43]:
```suggestion
    key = models.FloatField()
```

File: server/mesite/models.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    time_signature = models.IntegerField()
```
Comment (mezgoodle) [inline server/mesite/models.py:44]:
```suggestion
    time_signature = models.FloatField()
```

**Review by mezgoodle — CHANGES_REQUESTED** [review]
Fields changed

File: server/mesite/tests.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    @classmethod
+    def setUpTestData(cls):
+        Mood.objects.create(
+            mood='Happy',
+        )
+        mood = Mood.objects.get(id=1)
+
+        Song.objects.create(
+            name='Test name',
+            artist='Test artist',
```
Comment (mezgoodle) [inline server/mesite/tests.py:36]:
```suggestion
            artists='Test artist',
```

File: server/mesite/tests.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            acousticness=1.1,
+            danceability=3123.321,
+            energy=0.000,
+            instrumentalness=-1323.1,
+            liveness=1.0,
+            valence=1.,
+            loudness=123.3,
+            speechiness=3123.123,
+            tempo=123.1,
+            key=3123,
```
Comment (mezgoodle) [inline server/mesite/tests.py:47]:
```suggestion
            key=3123.0,
```

File: server/mesite/tests.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            time_signature=0,
```
Comment (mezgoodle) [inline server/mesite/tests.py:48]:
```suggestion
            time_signature=0.0,
```

**Review by mezgoodle — CHANGES_REQUESTED** [review]

File: server/mesite/tests.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+    def test_song_content(self):
+        """
+        Test fields on equality and instance of some types
+        """
+        song = Song.objects.get(id=1)
+        self.assertEquals(song.mood, Mood.objects.get(id=1))
+        self.assertIsInstance(song.song_id, str)
+        self.assertEquals(song.name, 'Test name')
+        self.assertIsInstance(song.energy, float)
```
Comment (mezgoodle) [inline server/mesite/tests.py:60]:
```suggestion

```

File: server/mesite/tests.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        self.assertEquals(song.liveness, 1.)
+        self.assertIsInstance(song.key, int)
```
Comment (mezgoodle) [inline server/mesite/tests.py:62]:
```suggestion

```

Comment (mezgoodle) [inline server/mesite/tests.py:62]:
```suggestion
        self.assertIsInstance(song.key, float)
```

**Review by mezgoodle — APPROVED** [review]

**Review by VsIG-official — APPROVED** [review]
вот е грейт ворк Ю хев зєр

**Pull request merged**